dashboard/src/actions/overviewActions.js

```diff
@@ -290,17 +305,23 @@ export const editMetadata =
 
     const rIndex = data.findIndex((item) => item.resource_id === rId);
     data[rIndex][metadata] = value;
-    data[rIndex][CONSTANTS.IS_DIRTY] = true;
-    if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][
-        CONSTANTS.NAME_ERROR_MSG
-      ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
-    } else if (value.length === 0) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
-    } else {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+
+    if (metadata === CONSTANTS.NAME_KEY) {
+      if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][
+          CONSTANTS.NAME_ERROR_MSG
+        ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
+      } else if (value.length === 0) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
+      } else {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+      }
+      data[rIndex][CONSTANTS.IS_DIRTY][CONSTANTS.DATASET_NAME_KEY] = true;
+    } else if (CONSTANTS.SERVER_DELETION_KEY) {
```

Comment:
Shouldn't this be
```;
    } else if (metadata === CONSTANTS.SERVER_DELETION_KEY) {
```
😀  (I failed to notice this the first dozen times that I looked at it!)

Reply:
:roll_eyes: :